feat: misc UI/UX improvements

alexef opened the pull request:
Tiny bits of UI, documented below commit by commit:

1. (excluded) [6de4777](https://github.com/argoproj/argo-cd/pull/11769/commits/6de47771aea24d50c33baca8dd90bf6cf1a0a3f1) - 
  ~ReplicaSets and HPA expose their desired/available/ready replicas in node info~

2. [689a93d](https://github.com/argoproj/argo-cd/pull/11769/commits/689a93d105c82e547e45660d69b37c7fe647c48c) - fix orphaned background in list view
<img width="1080" alt="Screenshot 2023-01-25 at 22 58 19" src="https://user-images.githubusercontent.com/346576/214700811-d1217fd5-b4a8-4c57-a74f-506350000427.png">

3. [5d16def](https://github.com/argoproj/argo-cd/pull/11769/commits/5d16def23af092ae92af94105ae6c92bf9c27c2e) - change tree view sorting of rs (default is by name, now by revision, latest on top)

<img width="692" alt="Screenshot 2023-01-25 at 23 00 09" src="https://user-images.githubusercontent.com/346576/214701172-e7e0cada-43d2-442c-a9cd-ba49bcd959e0.png">

4. [2758d0a73b08342963fd11764a474fc7df339c56](https://github.com/argoproj/argo-cd/pull/11769/commits/2758d0a73b08342963fd11764a474fc7df339c56) - fix white space for large messages (app conditions and event list)

5. (updated)  [e41043d](https://github.com/argoproj/argo-cd/pull/11769/commits/e41043dd86c7211b65526ebcfb27ee8840a6b2ae) - fix pod tooltip; use the same logic as in tree view, leverage and show pod.info messages; exclude redundant node

<img width="333" alt="Screenshot 2023-02-22 at 03 42 18" src="https://user-images.githubusercontent.com/346576/220508024-8e85f277-4206-49ac-8b09-ee34b678fe8f.png">


6. [ee5349e](https://github.com/argoproj/argo-cd/pull/11769/commits/ee5349e287ad83d31963fa12ea71ebc171f4f340) - namespace copy to clipboard in app summary 

<img width="591" alt="Screenshot 2023-01-25 at 23 02 16" src="https://user-images.githubusercontent.com/346576/214701544-66c0e641-d372-4652-8ee8-bc2d10b5d2eb.png">

7. [7b66f6a](https://github.com/argoproj/argo-cd/pull/11769/commits/7b66f6a5e6c7924ffde8cd093574f850a8007d8a) - click on OutOfSync same as we can click on Last Sync
<img width="376" alt="Screenshot 2023-01-25 at 23 03 19" src="https://user-images.githubusercontent.com/346576/214701790-06c01f12-4acf-46f6-b3f0-a13fbcaae11e.png">

---
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

alexef commented:
@rbreeze if you can have a look, I bundled together UX fixes from day to day experience 🙏

alexef commented:
@schakrad has a [better fix for long events messages](https://github.com/argoproj/argo-cd/pull/12267) - no more triming, I will update this PR to drop the titles if that one gets merged first.

alexef commented:
@rbreeze merged with UI changes in master, fixed conflicts, appreciate if you check this out.

<img width="1294" alt="Screenshot 2023-02-11 at 22 10 36" src="https://user-images.githubusercontent.com/346576/218281352-ac04ada3-1e6a-4fa0-a194-f1187a80375b.png">

rbreeze commented on controller/cache/info.go:
```diff
@@ -386,3 +398,27 @@ func populateHostNodeInfo(un *unstructured.Unstructured, res *ResourceInfo) {
+	if rs.Spec.Replicas == nil {
+		replicas = fmt.Sprintf("-/%d/%d", rs.Status.ReadyReplicas, rs.Status.Replicas)
+	} else {
+		replicas = fmt.Sprintf("%d/%d/%d", *rs.Spec.Replicas, rs.Status.ReadyReplicas, rs.Status.Replicas)
```
I'm not sure there's a better solution, but IMO it's not good UX to have unlabeled numbers like this. At the very least we should document this somewhere

alexef replied:
yeah, I'm also not 100% committed on it, will exclude from this change set

rbreeze commented on ui/src/app/applications/components/application-pod-view/pod-view.tsx:
```diff
@@ -167,13 +167,20 @@ export class PodView extends React.Component<PodViewProps> {
                                                                                 <div>
                                                                                     {pod.metadata.name}
                                                                                     <div>Health: {pod.health}</div>
+                                                                                    <div>
```
While I agree we should display this information in the pod view, I think in the current format it's information overload. I'd like to have a better design for the tooltip (or pod view in general) for a separate PR.

alexef replied:
Updated the PR, extracting this tooltip to a separate component (so that we can properly style it in a future PR).

*Before*:

<img width="355" alt="Screenshot 2023-02-22 at 03 03 37" src="https://user-images.githubusercontent.com/346576/220507725-e54488b7-96ff-4089-ac56-3045ae4c7c3c.png">

*After*:

<img width="333" alt="Screenshot 2023-02-22 at 03 42 18" src="https://user-images.githubusercontent.com/346576/220507759-fa844aba-0b5a-49e4-8b2f-e76cacfa0b62.png">

rbreeze replied:
Looks great!

rbreeze reviewed (changes requested):
Thank you for the improvements! Added a couple comments

In the future, it's easier as a reviewer to review many small PRs rather than one large one like this. Smaller PRs have the added benefit of being more clear in the commit log about what was actually changed. This PR is great, just a suggestion for next time! 🙂

alexef commented:
> In the future, it's easier as a reviewer to review many small PRs rather than one large one like this

I agree with you fully. I just got caught by the wave and considered them small, but once put together it ended up like this.

alexef commented:
@rbreeze thanks for the feedback. Updated the PR:
- removed the pod count labels for RS/HPA, will keep the idea for later
- updated the busy pod view tooltip, extracted into a separate file, removed node, used left align which makes it a bit more readable

alexef commented:
@rbreeze build is green again, ready to merge!

rbreeze reviewed (approved):
LGTM!

nobuyo commented:
@alexef Hi, can you take a look at the failed test? I'd love to see these improvements released.

alexef commented:
@nobuyo thanks for the ping. the failing test is unrelated to my changes. I've merged master into the branch, hopefully the build will pass now
